openhands/runtime/action_execution_server.py

```diff
@@ -374,9 +374,12 @@ async def validation_exception_handler(
     @app.middleware('http')
     async def one_request_at_a_time(request: Request, call_next):
         assert client is not None
-        async with client.lock:
-            response = await call_next(request)
-        return response
+        # Only apply lock for execute_action endpoint
+        if request.url.path == '/execute_action':
```

Comment:
oh wow why do we even have this middleware? why not allow multiple execute actions?

Reply:
If we have multiple *bash* execution requests going in simultaneously, it can cause issues (e.g., pexcept parsing) - or maybe we should really handle this inside action execution server

Reply:
Just pushed a new commit - now we do this inside client.run_action

Reply:
I suppose it really depends on what the action is doing, and there isn't really a way of determining if the action is safe or not.